common/client.go

```diff
@@ -172,20 +173,21 @@ func (rc *RestClient) modifyUrl(dest string, queryMod url.Values) error {
 
 // execMethod applies the specified method to the provided url (which is interpreted
 // as relative or absolute).
+// POST methods may not be idempotent, so for retry capability we will employ the following logic:
```

Comment:
Basically as agent now on a request returns "OK and then starts to do work, it reminded me it's a good idea to actually return 204, so I added some enhancements to make that possible and in the process I thought that this comment is better done as godoc.